muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java

```diff
@@ -1045,4 +1086,175 @@ else if (value instanceof Double) {
+      workspaceUserEntityController.unarchiveWorkspaceUserEntity(workspaceUserEntity);
+    }
+    if (workspaceUserEntity != null && workspaceUserEntity.getActive() == Boolean.FALSE) {
+      workspaceUserEntityController.updateActive(workspaceUserEntity, Boolean.TRUE);
+      userIndexer.indexUser(workspaceUserEntity.getUserSchoolDataIdentifier().getUserEntity());
+    }
+
+    fi.otavanopisto.muikku.schooldata.entity.WorkspaceUser workspaceUser = workspaceController.findWorkspaceUserByWorkspaceAndUser(workspaceIdentifier, studentEntity.defaultSchoolDataIdentifier());
+    if (workspaceUser == null) {
+      workspaceUser = workspaceController.createWorkspaceUser(workspace, student, WorkspaceRoleArchetype.STUDENT);
+      waitForWorkspaceUserEntity(workspaceEntity, studentIdentifier);
+    }
+    else {
+      workspaceController.updateWorkspaceStudentActivity(workspaceUser, true);
+    }
+
+    List<WorkspaceUserEntity> workspaceTeachers = workspaceUserEntityController.listActiveWorkspaceStaffMembers(workspaceEntity);
+    List<UserEntity> recipients = new ArrayList<UserEntity>();
+
+    String workspaceName = workspace.getName();
+    if (!StringUtils.isBlank(workspace.getNameExtension())) {
+      workspaceName += String.format(" (%s)", workspace.getNameExtension()); 
+    }
+
+    String userName = student.getNickName() == null
+      ? student.getDisplayName()
+      : String.format("%s \"%s\" %s (%s)", student.getFirstName(), student.getNickName(), student.getLastName(), student.getStudyProgrammeName());
+
+    User loggedUser = userController.findUserByUserEntityDefaults(sessionController.getLoggedUserEntity());
+
+    String loggedUserName = loggedUser.getNickName() == null
+        ? loggedUser.getDisplayName()
+        : String.format("%s \"%s\" %s (%s)", loggedUser.getFirstName(), loggedUser.getNickName(), loggedUser.getLastName(), loggedUser.getStudyProgrammeName());
+
+    recipients.add(studentEntity);
+
+    for (WorkspaceUserEntity workspaceTeacher : workspaceTeachers) {
+      recipients.add(workspaceTeacher.getUserSchoolDataIdentifier().getUserEntity());
+    }
+
+    UserSchoolDataIdentifier userSchoolDataIdentifier = userSchoolDataIdentifierController.findUserSchoolDataIdentifierBySchoolDataIdentifier(studentIdentifier);
+
+    workspaceController.createWorkspaceUserSignup(workspaceEntity, userSchoolDataIdentifier.getUserEntity(), new Date(), entity.getMessage());
+
+    String caption = localeController.getText(sessionController.getLocale(), "rest.workspace.joinWorkspace.joinNotification.counselor.caption");
+    caption = MessageFormat.format(caption, loggedUserName, userName, workspaceName);
+
+    String workspaceLink = String.format("<a href=\"%s/workspace/%s\" >%s</a>", baseUrl, workspaceEntity.getUrlName(), workspaceName);
+
+    String studentLink = String.format("<a href=\"%s/guider#userprofile/%s\" >%s</a>", baseUrl, studentIdentifier.toId(), userName);
```

Comment:
Does this mean that the student also gets a link to guider (which they cannot access)? Student seems to be in the recipient list at least.